source/adapters/hip/memory.hpp

```diff
@@ -196,4 +488,21 @@ struct ur_mem_handle_t_ {
   uint32_t decrementReferenceCount() noexcept { return --RefCount; }
 
   uint32_t getReferenceCount() const noexcept { return RefCount; }
+
+  void setLastEventWritingToMemObj(ur_event_handle_t NewEvent) {
+    assert(NewEvent && "Invalid event!");
+    // We only need this functionality for multi device context
+    if (Context->NumDevices == 1) {
```

Comment:
Would it be worth asserting on this, or is it not a failure state?

Reply:
I think good to add an assert. Thanks